config/middlewares/authorization.js

```diff
@@ -1,33 +1,44 @@
+import jwt from 'jsonwebtoken';
+import config from '../env/all';
+
+const { secret } = config;
+
 /**
  * Generic require login routing middleware
+ * @export
+ * @param {object} req
+ * @param {object} res
+ * @param {object} next
+ * @returns {object} response object
  */
-exports.requiresLogin = function(req, res, next) {
-    if (!req.isAuthenticated()) {
-        return res.send(401, 'User is not authorized');
-    }
-    next();
-};
+export function requiresLogin(req, res, next) {
+  const token = req.body.token || req.headers['x-token'] || req.headers.token;
+  if (token) {
+    jwt.verify(token, secret, (err, decoded) => {
+      if (err) {
+        res.status(401).send({
+          message: 'Token expired'
+        });
+      } else {
+        req.decoded = decoded;
+        next();
+      }
+    });
+  } else {
+    res.status(403).send({
+      message: 'Token not provided'
+    });
+  }
+}
```

Comment:
Good job here. Any reason why the requiresLogin function is not an arrow function?